### PR TITLE
activity logs added

### PR DESCRIPTION
**Pull request by saravanpa-aot**
*Issue #:*
https://github.com/bcgov/entity/issues/6739

*Description of changes:*

Changes for publishing activity logs
The nats changes to receive the activity logs wil be in a separate PR

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-api/src/auth_api/services/affiliation.py [hunk 1]
```diff
@@ -167,7 +168,7 @@ def create_affiliation(org_id, business_identifier, pass_code=None, token_info:
         affiliation.save()
 
         entity.set_pass_code_claimed(True)
-
+        publish_activity(ActivityAction.CREATE_AFFILIATION.value, 'BUSINESS', entity.name, entity_id)
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/services/affiliation.py:171]: Pass in org_id, how do we know which account this business got affiliated ?

File: auth-api/src/auth_api/services/entity.py [hunk 2]
```diff
@@ -31,7 +31,9 @@
 from auth_api.utils.passcode import passcode_hash
 from auth_api.utils.roles import ALL_ALLOWED_ROLES, Role
 from auth_api.utils.util import camelback2snake
+from .activity_log_publisher import publish_activity
 from .authorization import check_auth
+from ..utils.enums import ActivityAction
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/services/entity.py:36]: relative import, we need to put some standards, either allow relative import nor not allow at all. In pay and legal-api, relative import is not allowed.

File: auth-api/src/auth_api/services/entity.py [hunk 3]
```diff
@@ -254,3 +262,4 @@ def delete(self):
             self.delete_contact()
 
         self._model.delete()
+        # publish_activity(ActivityAction.REMOVE_AFFILIATION.value, 'BUSINESS', self._model.business_identifier)
```
Comment (sumesh-aot) [inline auth-api/src/auth_api/services/entity.py:265]: Why is this commented out ?

File: auth-api/migrations/versions/d0392ebda924_activity_stream.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    sa.Column('id', sa.Integer(), nullable=False),
+    sa.Column('actor', sa.String(length=250), nullable=True),
+    sa.Column('action', sa.String(length=250), nullable=True),
+    sa.Column('item_type', sa.String(length=250), nullable=True),
+    sa.Column('item_name', sa.String(length=250), nullable=True),
+    sa.Column('item_id', sa.String(length=250), nullable=True),
+    sa.Column('remote_addr', sa.String(length=250), nullable=True),
+    sa.Column('created_by_id', sa.Integer(), nullable=True),
+    sa.Column('modified_by_id', sa.Integer(), nullable=True),
+    sa.ForeignKeyConstraint(['created_by_id'], ['users.id'], ),
```
[the remark this answers is not in the document]
Comment (sumesh-aot) [reply auth-api/migrations/versions/d0392ebda924_activity_stream.py:34]: Yea, techdebt sounds good.. this is just adding more indexes and constraints which are not really needed.

[the remark this answers is not in the document]
Comment (saravanpa-aot) [reply auth-api/migrations/versions/d0392ebda924_activity_stream.py:34]: ok..will remove them constrains/indexes,thats a good point

**Review by sumesh-aot — APPROVED** [review]

**Pull request merged**